src/components/editor-page/editor-pane/linter/frontmatter-linter.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import type { Linter } from './linter'
+import type { EditorView } from '@codemirror/view'
+import type { Diagnostic } from '@codemirror/lint'
+import { extractFrontmatter } from '../../../../redux/note-details/frontmatter-extractor/extractor'
+import { load } from 'js-yaml'
+import type { RawNoteFrontmatter } from '../../../../redux/note-details/raw-note-frontmatter-parser/types'
+import { t } from 'i18next'
+
+/**
+ * Creates a {@link Linter linter} for the yaml frontmatter.
+ */
+export class FrontmatterLinter implements Linter {
+  lint(view: EditorView): Diagnostic[] {
+    const lines = view.state.doc.toString().split('\n')
+    const frontmatterExtraction = extractFrontmatter(lines)
+    if (!frontmatterExtraction.isPresent) {
+      return []
+    }
+    const frontmatterLines = lines.slice(0, frontmatterExtraction.lineOffset + 1)
+    const rawNoteFrontmatter = this.loadYaml(frontmatterExtraction.rawText)
+    if (rawNoteFrontmatter === undefined) {
+      // Could not parse frontmatter
```

Comment:
Shouldn't that be obvious? At this point the method hasn't found a frontmatter.

Reply:
I just wanted to give the magic undefined a bit more meaning without needing to look into loadYaml, but if you prefer I can leave it as is